app/questionnaire_state/relationship_state_question.py

```diff
@@ -11,13 +12,26 @@ def __init__(self, id, schema_item):
     def build_repeating_state(self, user_input):
         template_answer = self.answers.pop()
         group_instance = template_answer.group_instance
-        household_answers = get_answer_store(current_user).filter(answer_id='household-full-name')
-        sorted(household_answers, key=lambda household_answer: household_answer['answer_instance'])
-        remaining_answers = household_answers[group_instance + 1:] if self.group_instance < len(household_answers) else []
+        first_names = get_answer_store(current_user).filter(answer_id='first-name')
+        last_names = get_answer_store(current_user).filter(answer_id='last-name')
+
+        household_answers = list(zip(first_names, last_names))
+        remaining_answers = list(zip(first_names[group_instance + 1:], last_names[group_instance + 1:])) \
+            if self.group_instance < len(household_answers) else []
 
         for index, answer in enumerate(remaining_answers):
             for answer_schema in self.schema_item.answers:
                 new_answer_state = self.create_new_answer_state(answer_schema, index, group_instance)
-                new_answer_state.schema_item.widget.current_person = household_answers[group_instance]['value']
-                new_answer_state.schema_item.widget.other_person = answer['value']
+
+                current_person_name = format_household_member_name([
+                    household_answers[group_instance][0]['value'],
+                    household_answers[group_instance][1]['value'],
+                ])
+                new_answer_state.schema_item.widget.current_person = current_person_name
+
+                other_person_name = format_household_member_name([
+                    answer[0]['value'],
+                    answer[1]['value'],
```

Comment:
The 'hard-coded' 0 and 1 makes this not very readable. It might be clearer to create an array of dicts for the remaining people e.g.
```
remaining_people = [ 
  { 'first-name': 'Dave', 'last-name': 'Davis' }, 
  { 'first-name': 'Mike', 'last-name': 'Jones' },
  ... 
]
```